.copilot/cpp_instructions.md

```diff
@@ -0,0 +1,88 @@
+This document provides specific coding standards and best practices for C++ development within this repository. Adhering to these rules ensures code consistency, maintainability, and high performance.
+
+# 1. Coding Style and Formatting
+
+* **Auto-Formatting:** All C++ code must be formatted using the `.clang-format` file located in the repository root. This is a non-negotiable rule.
+* **Line Length:** The preferred line length is 120 characters, with a hard maximum of 160 characters for complex statements or function signatures to maintain readability.
+* **Brace Style:** Opening braces (`{`) should be on the same line as the control structure or function definition.
+* **Indentation:** Use 4 spaces for indentation.
+
+# 2. Naming Conventions
+
+* **`snake_case`:** All identifiers, including classes, structs, enums, functions, variables, member variables, and namespaces, must be in `snake_case`.
+    * **Example:** `my_class_name`, `http_request`, `local_variable`.
+* **Constants:** Global constants and `constexpr` variables should also be in `snake_case`. While `UPPER_SNAKE_CASE` may be present in older code, `snake_case` is the preferred convention for new code.
+    * **Example:** `static constexpr int max_connections = 100;`
+* **File Naming:** Header files should have a `.hh` extension and source files a `.cc` extension. Header file names should match the main class or module they contain (e.g., `http_server.hh` for `class http_server`).
+
+#### 3\. Modern C++ Usage
+
+* **C++ Standard:** The codebase uses C++20 or newer. Prefer features from these standards, especially coroutines, as they improve code clarity and safety.
+* **Resource Management:** Use Seastar-specific smart pointers and resource management types.
+    * Prefer `foreign_ptr` for sharing resources between Seastar and other code.
+    * Use `lw_shared_ptr` and `shared_ptr` for shared ownership within the Seastar framework.
+    * Avoid `std::unique_ptr`, `std::shared_ptr`, and raw pointers unless there is a clear, documented reason (e.g., interfacing with C-style APIs).
+* **Headers:**
+    * Use `#pragma once` for include guards.
+    * Include project headers first, followed by third-party library headers, and finally standard library headers.
+* **No `using namespace std;`:** Never include `using namespace std;` in header files. In source files, use it sparingly and only within function scope to minimize naming conflicts.
```

Comment:
While at it, you can tell the bot that the "using namespace seastar" is always done and can be assumed. It can usually be assumed because all our header files include seastarx.hh, which contains "using namespace seastar".

Reply:
It's like to move off it, and only import very common names like `future`.

Reply:
Yes, I also prefer to explicitly use "seastar" (this is what I recommended in seastar/doc/tutorial.md), but in the Scylla source code we're already hooked on this "using namespace seastar" ;-)